### PR TITLE
Reorganize how exposing and consuming connection data

### PR DESCRIPTION
**Pull request by retrofox**
This PR reorganizes the code among Connect, JS Connect, and the Protect plugin in order to make the API of some helpers and hooks simpler, making the focus on the recent `useProductCheckoutWorkflow()` custom hook.

Follow-up from https://github.com/Automattic/jetpack/pull/24122#discussion_r860154635

Fixes #

#### Changes proposed in this Pull Request:

*

#### Jetpack product discussion

p1651081554136539-slack-C02TQF5VAJD

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Test with Protect plugin
* Connect the site
* You should see the Footer section, with the `Get Jetpack Security` button there
* Confirm the browser leads you properly to the checkout page. It will try to log in the user in case it isn't already logged.
* It should go back to the Protect primary page
* Also you can take a look at the URL, trying to confirm it's well-formed.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202183659789068

### CODE REVIEW

**Comment by kraftbj** [comment]
The linting error is unrelated; will merge with master once #24141 has merged.

**Review by leogermani — CHANGES_REQUESTED** [review]

File: projects/packages/connection/src/class-initial-state.php
Comment (retrofox) [reply projects/packages/connection/src/class-initial-state.php:41]: my mistake

**Review by leogermani — APPROVED** [review]

**Pull request merged**
